tests/integration/test_disk_over_web_server/test.py

```diff
@@ -322,3 +322,74 @@ def test_replicated_database(cluster):
 
     node1.query("DROP DATABASE rdb SYNC")
     node2.query("DROP DATABASE rdb SYNC")
+
+
+def test_page_cache(cluster):
+    node = cluster.instances["node2"]
+    global uuids
+    assert len(uuids) == 3
+    for i in range(3):
+        node.query(
+            """
+            CREATE TABLE test{} UUID '{}'
+            (id Int32) ENGINE = MergeTree() ORDER BY id
+            SETTINGS storage_policy = 'web';
+        """.format(
+                i, uuids[i], i, i
+            )
+        )
+
+        result1 = node.query(
+            f"SELECT sum(cityHash64(*)) FROM test{i} SETTINGS use_page_cache_for_disks_without_file_cache=1 -- test cold cache"
+        )
+        result2 = node.query(
+            f"SELECT sum(cityHash64(*)) FROM test{i} SETTINGS use_page_cache_for_disks_without_file_cache=1 -- test warm cache"
+        )
+        result3 = node.query(
+            f"SELECT sum(cityHash64(*)) FROM test{i} SETTINGS use_page_cache_for_disks_without_file_cache=0 -- test no cache"
+        )
+
+        assert result1 == result3
+        assert result2 == result3
+
+        node.query("SYSTEM FLUSH LOGS")
+
+        def get_profile_events(query_name):
+            print(f"asdqwe {query_name}")
+            text = node.query(
+                f"SELECT ProfileEvents.Names, ProfileEvents.Values FROM system.query_log ARRAY JOIN ProfileEvents WHERE query LIKE '% -- {query_name}' AND type = 'QueryFinish'"
+            )
+            res = {}
+            for line in text.split("\n"):
+                if line == "":
+                    continue
+                name, value = line.split("\t")
+                print(f"asdqwe {name} = {int(value)}")
+                res[name] = int(value)
+            return res
+
+        ev1 = get_profile_events("test cold cache")
+        assert ev1.get("PageCacheChunkMisses", 0) > 0
+        assert (
+            ev1.get("DiskConnectionsCreated", 0) + ev1.get("DiskConnectionsReused", 0)
+            > 0
+        )
+
+        ev2 = get_profile_events("test warm cache")
+        assert ev2.get("PageCacheChunkDataHits", 0) > 0
+        assert ev2.get("PageCacheChunkMisses", 0) == 0
+        assert (
+            ev2.get("DiskConnectionsCreated", 0) + ev2.get("DiskConnectionsReused", 0)
```

Comment:
@al13n321 this check fails from time to time, can you please take a look?
https://play.clickhouse.com/play?user=play#U0VMRUNUIGNoZWNrX25hbWUsIHRlc3RfbmFtZSwgcmVwb3J0X3VybApGUk9NIGNoZWNrcwpXSEVSRSB0ZXN0X25hbWUgTElLRSAnJXRlc3RfcGFnZV9jYWNoZSUnCiAgICBBTkQgY2hlY2tfc3RhcnRfdGltZSA+PSBub3coKSAtIElOVEVSVkFMIDcgZGF5Ci8vICAgIEFORCBwdWxsX3JlcXVlc3RfbnVtYmVyID0gMAogICAgQU5EIHRlc3Rfc3RhdHVzICE9ICdTS0lQUEVEJwogICAgQU5EIHRlc3Rfc3RhdHVzIExJS0UgJ0YlJwogICAgQU5EIGNoZWNrX3N0YXR1cyAhPSAnc3VjY2VzcycKT1JERVIgQlkgY2hlY2tfbmFtZSwgdGVzdF9uYW1lLCBjaGVja19zdGFydF90aW1l